classes/target/target_dataroot.php

```diff
@@ -116,7 +116,8 @@ protected function load_from_array($array) {
 
         $result = true;
 
-        $target = rtrim($this->get_full_path(), '/') . '/' . $this->settings['filename'];
+        $fullpath = $this->get_full_path();
+        $target = rtrim($fullpath, '/') . '/' . $this->get_target_file_name($fullpath . $this->settings['filename']);
```

Comment:
how about to abstract this code to a new function and make it reusable in the future?